### PR TITLE
chore: use templates from default repository

### PR DESCRIPTION
**Pull request by kayman-mk**
This PR syncs all templates from the default `Repository-Template-Terraform-Module.`

### CODE REVIEW

**Comment by kayman-mk** [comment]
We can go with the failing pipeline. This will be fixed by using the new workflow files.

File: .github/update_templates.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+ensure_dependencies_or_exit
+ensure_repo_preconditions_or_exit
+
+latest_template_path=$(mktemp -d -t repository-template-XXXXX)
+new_branch_name=$(basename "$latest_template_path")
+
+# clone the default branch to get the latest version of the template files
+gh repo clone https://github.com/Hapag-Lloyd/Repository-Template-Maven.git "$latest_template_path"
```
Comment (PascalFrenz) [inline .github/update_templates.sh:55]: This references the wrong repo. Should be https://github.com/Hapag-Lloyd/Repository-Template-Terraform-Module, right?

**Review by PascalFrenz — CHANGES_REQUESTED** [review]

**Review by PascalFrenz — APPROVED** [review]

**Comment by PascalFrenz** [comment]
@kayman-mk can these changes be merged?

**Pull request merged**